### ISSUE TITLE
linter: nextjs/no-html-link-for-pages should not flag anchor tags with target="_blank"

### ISSUE THREAD

**Issue by milehighideas** [oxc-project/oxc#18676]
## Description

The `nextjs/no-html-link-for-pages` rule currently flags all `<a>` tags linking to pages, even when they have `target="_blank"`. This is a false positive — `next/link` should not be used for links that open in a new tab, as the client-side navigation behavior is not desired in that case.

## Expected behavior

Anchor tags with `target="_blank"` should be exempt from this rule, matching the behavior of the upstream ESLint plugin.

## Example (false positive)

```jsx
// This should NOT trigger the rule — it's an external link opening in a new tab
<a href={config.docsUrl} target="_blank" rel="noopener noreferrer">
  Get your API key
</a>
```

## Upstream fix

This was fixed in the original Next.js ESLint plugin in December 2021:

- **Issue**: https://github.com/vercel/next.js/issues/28547
- **PR**: https://github.com/vercel/next.js/pull/32780

The fix checks for `target="_blank"` on the `<a>` element and skips the lint warning when present.

## Current oxlint source

The current implementation in `crates/oxc_linter/src/rules/nextjs/no_html_link_for_pages.rs` only inspects the `href` attribute and has no logic for checking `target` or `_blank`.

## Environment

- oxlint v1.42.0
- Next.js 16.1.1
